File: 724/main.go
package main 

import "fmt"

func pivotIndex(nums []int) int {
	ans := 0 
	t := 0
	for _, v := range nums {
		t += v
	}

	for i, v := range nums {
		if 2 * ans + v == t {
			return i
		}
		ans += v
	}
	return -1
}

func main() {
	fmt.Println(pivotIndex([]int{1,7,3,6,5,6}))
	fmt.Println(pivotIndex([]int{1,2,3}))
	fmt.Println(pivotIndex([]int{2,1,-1}))

}

File: 724/main.rs
struct Solution {}

impl Solution {
    pub fn pivot_index(nums: Vec<i32>) -> i32 {
        let mut t: i32 = 0;
        for i in 0..nums.len() {
            t += nums[i];
        }

        let mut s: i32 = 0;
        for (i, &v) in nums.iter().enumerate() {
            if 2 * s + v == t {
                return i as i32
            }
            s += v;
        }
        -1
    }
}

fn main() {
    println!("{}", Solution::pivot_index(vec![1,7,3,6,5,6]))
}